src/builder/value_parser.rs

```diff
@@ -1051,6 +1079,136 @@ where
     }
 }
 
+#[derive(Copy, Clone, Debug)]
+pub struct InRange<P>
+where
+    P: TypedValueParser,
```

Comment:
Side note: this won't work with `ValueParser`s.  I think that'll be ok, but wanted to call it out

Reply:
Yeah, I'm aware. I didn't see a good way around that.